storage/diemdb/src/event_store/mod.rs

```diff
@@ -238,6 +282,84 @@ impl EventStore {
             })
             .collect::<Result<Vec<_>>>()
     }
+
+    /// Finds the first event sequence number in a specified stream on which `comp` returns false.
+    /// (assuming the whole stream is partitioned by `comp`)
+    fn search_for_event_lower_bound<C>(
+        &self,
+        event_key: &EventKey,
+        mut comp: C,
+        ledger_version: Version,
+    ) -> Result<Option<u64>>
+    where
+        C: FnMut(&ContractEvent) -> Result<bool>,
+    {
+        let mut begin = 0u64;
+        let mut end = match self.get_latest_sequence_number(ledger_version, event_key)? {
```

Comment:
why you no like map_or_else

Reply:
The None branch quits this method, `map_or_else()` doesn't do that?